Automattic/wp-calypso
# Cancel Purchase Form: Provide more context about product being canceled

Resolves to #73908 

## Proposed Changes

* Currently the cancel purchase form displays the `site.slug` prop next to the "Remove product" or "Remove plan" heading at the top of the page. This means that regardless of the product being removed, whether it's a plan, domain connection for a secondary domain, or add-on product such as a site redirect or custom CSS, the only thing being displayed is the primary site address.

This PR seeks to provide additional context for the product/plan being removed. I decided to use switch statement here on the condition of `purchase.productId` matching a domain connection, site redirect, and a default description which should suffice for plans or other add-ons.

## Before

### Any non-plan Product Removal - Before

![01 - before - removing any non-plan product](https://user-images.githubusercontent.com/65082164/222604173-be90e9ea-763e-4600-b167-6e90757014b2.png)

### Plan Removal - Before

![02 - before - removing any plan](https://user-images.githubusercontent.com/65082164/222604271-f0978fa0-1b8c-4a07-9601-8b1115f1e386.png)

## After 

### Removing domain connection that is not the primary address
Uses the `purcachase.meta` prop to display the name of the actual domain connection

![03 - after - removing domain connection](https://user-images.githubusercontent.com/65082164/222604577-21ace247-993d-41db-b61c-4b0af9fd25ba.png)


### Removing site redirect
Uses the `purcachase.meta` prop to display the name of the domain that the site redirect is pointing to
![04 - after - removing site redirect](https://user-images.githubusercontent.com/65082164/222604741-32f15ab6-1af3-4ffe-8eaf-75d21f30dc38.png)


### Removing any non-plan product, such as custom CSS
![05 - after - removing any other product](https://user-images.githubusercontent.com/65082164/222604917-3411ef89-f0a7-4c59-acc3-02fdc5d3d730.png)

### Removing any plan

![06 - after - removing any plan](https://user-images.githubusercontent.com/65082164/222605042-cfad8b68-becd-4c15-a676-ccbc1d18d47e.png)


## Testing Instructions



* Add a plan, domain connection, and add-ons including custom CSS and Site Redirect
* Remove  each plan and ensure that the context displayed at the top reflects the name of the product being removed and the appropriate address is being displayed that the removal is affecting
* Test on Simple & Atomic

## Pre-merge Checklist
